src/types.ts

```diff
@@ -347,11 +352,11 @@ export interface IModelOptions {
 export interface ICustomOptions {
   /**
    * Set the modelName of the class
-   *
+   * if it is a function, run the function.
    * if "automaticName" is true it sets a *suffix* instead of the whole name
```

Comment:
maybe also add an note that `customName` overwrites no matter what if its an function